src/server/json_family.cc

```diff
@@ -588,8 +589,101 @@ OpResult<long> OpClear(const OpArgs& op_args, string_view key, string_view path)
   return clear_items;
 }
 
+// Returns string vector that represents the pop out values.
+OpResult<vector<OptString>> OpArrPop(const OpArgs& op_args, string_view key, string_view path,
+                                     int index) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptString> vec;
+  auto cb = [&](const string& path, json& val) {
+    if (!val.is_array() || val.empty()) {
+      vec.emplace_back(nullopt);
+      return;
+    }
+
+    size_t removal_index;
+    if (index < 0) {
+      int temp_index = index + val.size();
+      removal_index = abs(temp_index);
+    } else {
+      removal_index = index;
+    }
+
+    if (removal_index >= val.size()) {
+      removal_index %= val.size();  // rounded to the array boundaries.
+    }
+
+    size_t i = 0;
+    for (auto it = val.array_range().begin(); it != val.array_range().end(); it++) {
+      if (i == removal_index) {
+        ostringstream os;
```

Comment:
I wonder - is it the only way to convert elements into strings? i.e. only with ostringstream ?

Reply:
It's possible to dump JSON object to an object that implements `std::basic_ostream<char_type>`